Sources/Plasma/PubUtilLib/plGImage/plPNG.cpp

```diff
@@ -226,27 +229,63 @@ bool plPNG::IWrite(plMipmap* source, hsStream* outStream)
             row_ptrs[i] = (png_bytep)srcp + (i * stride);
         }
 
+        // Write image data
         png_write_image(png_ptr, row_ptrs);
+
+        // Prepare Textual Metadata
+        char time_string[26]; // RFC-1123: "Day, DD-Mon-YYYY hh:mm:ss";
+        time_t rawtime = time(nullptr);
+        strftime(time_string, sizeof(time_string), "%a, %d-%b-%Y %T", gmtime(&rawtime));
+
+        // Add our standard fields, then combine with the custom fields
+        std::multimap<ST::string, ST::string> all_fields;
+        all_fields.emplace("Software", plProduct::ProductString());
+        all_fields.emplace("Creation Time", time_string);
+        for (auto field : textFields)
+            all_fields.insert(field);
+
+        png_text* text = new png_text[all_fields.size()];
+        size_t num_txtfields = 0;
+        for (auto it = all_fields.begin(); it != all_fields.end(); it++, num_txtfields++) {
+            text[num_txtfields].key = (png_charp)(strdup(it->first.left(79).to_utf8().data()));
+            text[num_txtfields].text = (png_charp)(strdup(it->second.to_utf8().data()));
```

Comment:
Maybe @zrax will know better, but I feel like we don't need to `strdup` at minimum `it->second` because we're just using the ST::string's UTF-8 buffer... Isn't the ST::string already internally UTF-8, so this buffer is guaranteed to not go out of scope? I'm a little uncomfortable with `strdup`'ing `it->first`, but I don't see a reasonable way around it.

Reply:
I think you're correct.  I've simplified this, pending further corrections from @zrax.